packages/flutter/lib/src/widgets/shortcuts.dart

```diff
@@ -436,6 +459,37 @@ class Shortcuts extends StatefulWidget {
     return inherited?.notifier;
   }
 
+  /// Returns the [Intent] matching the given key set.
+  ///
+  /// Returns an intent matching a set of pressed keys, as if a key had been
+  /// propagated to all of the [Shortcuts] widgets in the given context.  If
+  /// `keySet` is not supplied, uses the current set of pressed keys in
+  /// [RawKeyboard.keysPressed].
+  ///
+  /// This allows determining ahead of time which intent would be matched from a
+  /// set of pressed keys.
+  ///
+  /// Returns null if it reaches the root of the widget tree and no [Shortcuts]
+  /// widget's [ShortcutManager] has a matching [Intent] for the key set.
+  static Intent? matchingIntent(BuildContext context, {KeySet<KeyboardKey>? keySet}) {
+    assert(context != null);
+    InheritedElement? inherited = context.getElementForInheritedWidgetOfExactType<_ShortcutsMarker>();
+    while (inherited != null) {
+      final _ShortcutsMarker marker = inherited.widget as _ShortcutsMarker;
+      final ShortcutManager manager = marker.notifier!;
```

Comment:
I wonder if _ShortcutsMarker should override `notifier` to always return a non-null value since `_ShortcutsMarker` knows that it will always be non-null. The `!` here is a little surprising and it is not clear that notifier will always be non-null.

In _ShortcutsMarker something like

```dart
  @override
  ShortcutManager get notifier => super.notifier!;
```

Reply:
+1 that's even better!